content/blog/federating-spire-on-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,447 @@
+
+It requires the following federation configurations: 
+
+### C﻿luster Federated Trust Domain:
+
+![Cluster Federated Trust Domain](/img/table1.png)
+
+### B﻿undle Endpoint:
+
+![Bundle Endpoint](/img/table2.png)
+
+The sample CRD’s can be applied to each cluster.
+
+```shellsession
+#For K8s cluster 1:
+kubectl apply -f services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml 
+#For K8s cluster 2:
+kubectl apply -f services/spire/federation/clusterfederatedtrustdomain-cluster2.yaml 
+```
+
+The Bundle endpoint URL and the TrustDomainBundle fields must be edited to be configured to the specifications of your cluster. 
+
+Using the previous step, the Bundle endpoint URL can be obtained. To obtain the trust domain bundle, edit the clusters as follows: 
+
+```shellsession
+Cluster1:~ # kubectl exec -n spire -c spire-server deployment/spire-server -- /opt/spire/bin/spire-server bundle show -format spiffe > cluster1.bundle 
+
+Cluster2:~ # kubectl exec -n spire -c spire-server deployment/spire-server -- /opt/spire/bin/spire-server bundle show -format spiffe > cluster2.bundle 
+```
+
+Doing a cat command of the bundles reveals the keys to be copied into the trust domain bundle field.  
+
+After the Bundle endpoint URL and the TrustDomainBundle fields are configured and applied, check the status of the  SPIRE pods to make sure they are running and check the logs of the spire-server to verify successful federation.
+
+```shellsession
+Cluster1:~ # kubectl logs -n spire -c spire-server spire-server-574474c7dc-gbzl6 
+```
+
+![Cluster-1 Logs](/img/table3.png)
+
+```shellsession
+Cluster2:~ # kubectl logs -n spire -c spire-server spire-server-574474c7dc-2bfcx 
+```
+
+![Cluster-2 Logs](/img/table4.png)
+
+# Step 4. Deploying a sample application
+
+Now that SPIRE is federated and communication across clusters can be facilitated, here's how you can deploy a sample application that verifies this functionality.   
+
+## 4.1 Deploy a resource in Cluster-1
+
+In Cluster 1, apply a new ClusterSpiffeID called ***federated*** that registers resources with the label **spiffe.io/spire-managed-identity=curl-greeter** that can be federated with cluster2. Create a resource called ***curl-greeter*** that has the label: **spiffe.io/spire-managed-identity=curl-greeter** and annotation: **inject.istio.io/templates=sidecar, spire** 
+
+```shellsession
+kubectl run curl-greeter --image=radial/busyboxplus:curl --labels="spiffe.io/spire-managed-identity=curl-greeter" --overrides='{ "apiVersion": "v1", "spec": { "template": {"metadata": {"annotations": { "inject.istio.io/templates":"sidecar,spire" } } }}}' -i --tty 
+```
+
+## 4﻿.2 Deploy Bookinfo Sample Application in Cluster-2
+
+In Cluster 2, apply a new ClusterSpiffeID called ***federated*** that registers resources with the label **spiffe.io/spire-managed-identity=spire** that can be federated with cluster1. Apply the bookinfo sample application manifest. 
```

Comment:
For Cluster 2, you mention in Cluster 2, apply a new ClusterSpiffeID called federated that registers resouces with the label ....

Does it mean that one would need to apply the same kubectl command as in 4.1 before applying the Bookinfo sample app manifest in Cluster 2?
kubectl run curl-greeter --image=radial/busyboxplus:curl --labels="spiffe.io/spire-managed-identity=curl-greeter" --overrides='{ "apiVersion": "v1", "spec": { "template": {"metadata": {"annotations": { "inject.istio.io/templates":"sidecar,spire" } } }}}' -i --tty